src/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosure.java

```diff
@@ -499,11 +503,11 @@ private void findBackupFile(HttpMessage originalMessage) throws Exception {
 
 				//just to complicate things.. I have a test case which for the random file, does NOT give a 404 (so gives404s == false) 
 				//but for a "Copy of" file, actually gives a 404 (for some unknown reason). We need to handle this case.
-				if ( 	( gives404s && requestStatusCode != HttpStatus.SC_NOT_FOUND) || 
-						(	(!gives404s) &&
-								nonexistfilemsg .getResponseHeader().getStatusCode() != requestStatusCode &&
-							(! Arrays.equals(disclosedData, nonexistfilemsgdata)) 
-						)
+				if ( !isEmptyResponse(disclosedData) && ( gives404s && 
```

Comment:
Needs parentheses surrounding the right operand (to be `gives404s` and `!gives404s`).

Reply:
I mean `isEmptyResponse(disclosedData) && ( (gives404s...) || (!gives404s...) )` (some redundant parentheses removed for clarity).